src/core/Header/Header.tsx

```diff
@@ -0,0 +1,206 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import '@itwin/itwinui-css/css/header.css';
+import SvgChevronRight from '@itwin/itwinui-icons-react/cjs/icons/ChevronRight';
+import SvgHelpCircularHollow from '@itwin/itwinui-icons-react/cjs/icons/HelpCircularHollow';
+import SvgMoreVertical from '@itwin/itwinui-icons-react/cjs/icons/MoreVertical';
+import SvgNotification from '@itwin/itwinui-icons-react/cjs/icons/Notification';
+
+import cx from 'classnames';
+import React from 'react';
+import { IconButton } from '../Buttons';
+
+import { useTheme } from '../utils/hooks/useTheme';
+import { DropdownMenu } from '../DropdownMenu';
+
+export type HeaderTranslations = {
+  notifications: string;
+  help: string;
+  myAccount: string;
+  moreOptions: string;
+};
+
+export type HeaderProps = {
+  /**
+   * Icon to the application.
+   */
+  appIcon: React.ReactNode;
+  /**
+   * Name of the application.
+   */
+  appName?: React.ReactNode;
+  /**
+   * Handler on application click
+   */
+  onApplication?: () => void;
+  /**
+   * `HeaderButton` array, chevron will be added between each.
+   */
+  breadcrumbsButtons?: React.ReactNode[];
+  /**
+   * Content to display in the center section of the header, should handle iui-slim class if used.
+   */
+  centerContent?: React.ReactNode;
```

Comment:
Hmm I'm not sure about asking the user to handle a css class.

Also would call this `children` because it's not part of the main content we provide.

Reply:
I can switch to `children`, however it is put in a very specific place in the center, not like in the `Tile` where you can do whatever you want, but it needs to be aware of what happens when they select "slim"... otherwise it will do just like my example GIF where the searchbox overflow the header...

Reply:
I can switch to `children`, however it is put in a very specific place in the center, not like in the `Tile` where you can do whatever you want, but it needs to be aware of what happens when they select "slim"... otherwise it will do just like my example GIF where the searchbox overflow the header...

Reply:
Well, the user of this component is providing the `isSlim` prop, so they should know to handle the height manually

Reply:
children it is.